plugins/CuraProfileReader/CuraProfileReader.py

```diff
@@ -98,19 +99,23 @@ def _loadProfile(self, serialized: str, profile_id: str) -> Optional[InstanceCon
     #   \param profile_id The name of the profile.
     #   \param source_version The profile version of 'serialized'.
     #   \return List of serialized profile strings and matching profile names.
-    def _upgradeProfileVersion(self, serialized: str, profile_id: str, source_version: int) -> List[Tuple[str, str]]:
-        converter_plugins = PluginRegistry.getInstance().getAllMetaData(filter = {"version_upgrade": {} }, active_only = True)
+    def _upgradeProfileVersion(self, serialized: str, profile_id: str, main_version: int, setting_version: int) -> List[Tuple[str, str]]:
+        source_version = main_version * 1000000 + setting_version
 
-        source_format = ("profile", source_version)
-        profile_convert_funcs = [plugin["version_upgrade"][source_format][2] for plugin in converter_plugins
-                                 if source_format in plugin["version_upgrade"] and plugin["version_upgrade"][source_format][1] == InstanceContainer.Version]
+        from UM.VersionUpgradeManager import VersionUpgradeManager
+        results = VersionUpgradeManager.getInstance().updateFilesData("quality_changes", source_version, [serialized], [profile_id])
+        serialized = results.files_data[0]
 
-        if not profile_convert_funcs:
-            Logger.log("w", "Unable to find an upgrade path for the profile [%s]", profile_id)
+        parser = configparser.ConfigParser(interpolation = None)
+        parser.read_string(serialized)
+        if "general" not in parser:
+            Logger.log("w", "Missing required section 'general'.")
+            return []
+        if "version" not in parser["general"]:
+            Logger.log("w", "Missing required 'version' property")
```

Comment:
Instead of re-parsing the entire file, `updateFilesData` also just returns the version number you're trying to read. You can check it with:
```suggestion
        new_source_version = results.version
        if int(new_source_version / 1000000) != InstanceContainer.Version or new_source_version % 1000000 != CuraApplication.SettingVersion:
            Logger.log("e", "Failed to upgrade profile [%s]", profile_id)
```

(Note that this code is not tested. Please test first.)